### PR TITLE
Add timeout and retries to BigQuery entity import & rake task for run all required imports

### PR DESCRIPTION
**Pull request by despo**
## Context

Increase BigQuery timeout (default is 60) and add retries. This way any timeouts should be handled and reprocessed by BigQuery who handles and each retry by doubling the existing timeout.

Also introduces rake task that imports all require data to BigQuery as manual importing each table is very time consuming.

## Link to Trello card

https://trello.com/c/SljGT6Dl/4261-backfill-bigquery-data-from-apply

## Things to check

- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: lib/tasks/import_data_to_bigquery.rake [hunk 1]
```diff
@@ -1,7 +1,28 @@
 DEFAULT_BATCH_SIZE = 200
 SLEEP_TIME = 2
+ENTITY_MODELS = %w[Candidate ApplicationChoice ApplicationExperience ApplicationFeedback
```
Comment (duncanjbrown) [inline lib/tasks/import_data_to_bigquery.rake:3]: Could we map these from `analytics.yml` on the scheme of https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/spec/config/analytics_spec.rb?

File: app/workers/send_events_to_bigquery.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@ class SendEventsToBigquery
 
   def perform(request_event_json)
     table_name = ENV.fetch('BIG_QUERY_TABLE_NAME', 'events')
-    bq = Google::Cloud::Bigquery.new(project: ENV.fetch('BIG_QUERY_PROJECT_ID'))
+    bq = Google::Cloud::Bigquery.new(project: ENV.fetch('BIG_QUERY_PROJECT_ID'), retries: 3, timeout: 120)
```
Comment (cwrw) [inline app/workers/send_events_to_bigquery.rb:9]: 👍

File: lib/tasks/import_data_to_bigquery.rake [hunk 3]
```diff
@@ -1,7 +1,28 @@
...
+  desc 'Import all entity events to BigQuery'
+  task :import_all_entity_events, %i[sleep_time batch_size] do |_, args|
+    sleep_time = args.fetch(:sleep_time, SLEEP_TIME).to_i
+    batch_size = args.fetch(:batch_size, DEFAULT_BATCH_SIZE).to_i
+
+    ENTITY_MODELS.each do |model_name|
+      Rake::Task['bigquery:import_entity_events'].invoke(model_name,
+                                                         sleep_time,
+                                                         batch_size)
+      Rake::Task['bigquery:import_entity_events'].reenable
```
Comment (cwrw) [inline lib/tasks/import_data_to_bigquery.rake:22]: TIL

File: lib/tasks/import_data_to_bigquery.rake (hunk 1 above)
Comment (despo) [reply lib/tasks/import_data_to_bigquery.rake]: Updated.

File: lib/tasks/import_data_to_bigquery.rake [hunk 4]
```diff
@@ -1,7 +1,32 @@
 DEFAULT_BATCH_SIZE = 200
 SLEEP_TIME = 2
+IRREGULAR_TABLE_NAMES = {
+  provider_relationship_permissions: 'ProviderRelationshipPermissions',
+  provider_user_notifications: 'ProviderUserNotificationPreferences',
+  provider_users_providers: 'ProviderPermissions',
+  references: 'ApplicationReference',
+}.freeze
```
Comment (duncanjbrown) [inline lib/tasks/import_data_to_bigquery.rake:8]: :+1: we can decide how to centralise this later

File: lib/tasks/import_data_to_bigquery.rake [hunk 5]
```diff
@@ -1,7 +1,32 @@
...
+
+def entity_models
+  Rails.configuration.analytics.deep_merge(Rails.configuration.analytics_pii).map do |table_name, _|
```
Comment (duncanjbrown) [inline lib/tasks/import_data_to_bigquery.rake:11]: No need to deep-merge here — analytics contains all fields and pii contains a subset

File: lib/tasks/import_data_to_bigquery.rake (hunk 4 above)
Comment (despo) [reply lib/tasks/import_data_to_bigquery.rake:8]: Yes. There is still a bit of refactoring to do, just need to test this in QA as my local data are not optimal and I was also getting some SSL and some unreachable host errors that I think may have been caused by my connection being a bit flaky.

**Review by cwrw — APPROVED** [review]

**Pull request merged**